packages/data-context/test/unit/sources/ProjectDataSource.spec.ts

```diff
@@ -442,6 +444,74 @@ describe('getPathFromSpecPattern', () => {
   })
 })
 
+describe('_makeSpecWatcher', () => {
+  let ctx: DataContext
+  let specWatcher: chokidar.FSWatcher
+
+  beforeEach(function () {
+    this.timeout(20000) // FixturesHelper.remove can take awhile depending on what you've done prior
+    FixturesHelper.remove()
+
+    ctx = createTestDataContext('open')
+
+    FixturesHelper.scaffold('spec-watcher')
+
+    this.specWatcherPath = FixturesHelper.projectPath('spec-watcher')
+
+    ctx.actions.project.setCurrentProjectAndTestingTypeForTestSetup(this.specWatcherPath)
+  })
+
+  afterEach(async () => {
+    sinon.restore()
+    await specWatcher.close()
+    ctx.destroy()
+  })
+
+  it('watch for changes on files based on the specPattern', async function () {
+    function delay (ms) {
+      return new Promise((resolve) => {
+        setTimeout(resolve, ms)
+      })
+    }
+
+    specWatcher = ctx.project._makeSpecWatcher({
+      projectRoot: this.specWatcherPath,
+      specPattern: ['**/*.{cy,spec}.{ts,js}'],
+      excludeSpecPattern: ['**/ignore.spec.ts'],
+      additionalIgnorePattern: ['additional.ignore.cy.js'],
+    })
+
+    await new Promise((resolve) => {
+      specWatcher.once('ready', resolve)
+    })
+
+    const allFiles = new Set()
+
+    specWatcher.on('add', (filePath) => {
+      allFiles.add(filePath)
+    })
+
+    specWatcher.on('change', (filePath) => {
+      allFiles.add(filePath)
+    })
+
+    const SUPPORT_FILE = path.join(this.specWatcherPath, 'cypress', 'support', 'e2e.js')
+    const SPEC_FILE1 = path.join(this.specWatcherPath, 'cypress', 'e2e', 'foo.cy.js')
+    const SPEC_FILE2 = path.join(this.specWatcherPath, 'cypress', 'e2e', 'some', 'new', 'folder', 'foo.cy.js')
+
+    ctx.actions.file.writeFileInProject(SUPPORT_FILE, '// foo')
+    ctx.actions.file.writeFileInProject(SPEC_FILE1, '// foo')
+    ctx.actions.file.writeFileInProject(SPEC_FILE2, '// foo')
+
+    await delay(500)
```

Comment:
This is bad practice for unit tests, it makes the tests really slow if we have delays like this. Is there anything you could `await` or use `done` instead?